src/Resources/views/Form/roles_matrix_list.html.twig

```diff
@@ -0,0 +1,20 @@
+{#
+
+This file is part of the Sonata package.
+
+(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+
+#}
+{% for custom_role, attr in roles %}
+    <li>{{ form_widget(attr.form) }}</li>
+    {% if attr.read_only %}
+        <script>
+            $(document).ready(function () {
+                $('input[value="{{ custom_role }}"]').iCheck('disable');
+            });
+        </script>
```

Comment:
Please create a new javascript file just for this, and use data attributes on `li` elements to tell whether this is read only or not.